staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go

```diff
@@ -744,6 +746,14 @@ func (s *store) GetList(ctx context.Context, key string, opts storage.ListOption
 		})
 		metrics.RecordEtcdRequest(metricsOp, s.groupResource, err, startTime)
 		if err != nil {
+			if errors.Is(err, etcdrpc.ErrFutureRev) {
+				currentRV, getRVErr := s.GetCurrentResourceVersion(ctx)
+				if getRVErr != nil {
```

Comment:
Do we need to surface this error somewhere, or just log it maybe?

Reply:
@wojtek-t Sounds good. Shall I create the issue, or do you want to create and assign it to me?

Reply:
Please go ahead and create it